googletest/include/gtest/gtest_pred_impl.h

```diff
@@ -75,7 +75,7 @@ namespace testing {
 #define GTEST_ASSERT_(expression, on_failure) \
   GTEST_AMBIGUOUS_ELSE_BLOCKER_ \
   if (const ::testing::AssertionResult gtest_ar = (expression)) \
-    ; \
+    ::testing::UnitTest::GetInstance()->increment_success_assert_count(); \
```

Comment:
This seems dangerous.
Is there any synchronization here?
If there is, it is going to make this slower.
If there isn't, it is going to make this thread-hostile.

Reply:
(this is from internal review)

Reply:
Good point,
 - the main purpose of the assert counting, and why I think it is important, is to be able to easily spot whether a test actually has tested something reasonably close to what the intentions are, so a 100% accuracy is not really critical, but it is of course better/easier if it is accurate, so this is a tradeoff.

Unless there's a way to detect whether a test is exposed to multithreading, I suppose there's then two main choices: 
a) Leave it as it is for performance reasons, and document that it is a hint.
b) Use an atomic operation for the counting, which at least is more light weight than a locking based scheme.

 It could possibly be made configurable to be able to disable it for particularly challenging tests.
I suggest to opt for b) - I would argue that getting this information is worth the extra cycle cost.

Reply:
There's of course always the remaining question of whether using an atomic will have undesired effects on the parallelism in the test, possibly affect test outcomes, but that can probably be both positive and negative depending on the purpose of the test.

Reply:
Has this been updated to use atomics?